### PR TITLE
[CoW] Fix Fee USD Calculation

### PR DESCRIPTION
**Pull request by bh2smith**
We recently discovered a HUGE precision/rounding error on the calculation of `fee_usd` in the case when sell_price is not null. Since the fees are taken in sell token, there is no reason for us to try to take the larger of the two. here.

That is, when sell_price is not null, then  is `fee_usd = sell_price * fee`.

cc @josojo for internal approval

In particular [this transaction](https://etherscan.io/tx/0x84d57d1d57e01dd34091c763765ddda6ff713ad67840f39735f0bf0cced11f02)

I will link the issue with the old calculation inline near the code change.

Included also is a [PoC query](https://dune.com/queries/1841840) showing what the new fee calculation will look like in comparison.

### CODE REVIEW

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 1]
```diff
@@ -205,16 +205,10 @@ valued_trades as (
            fee,
            fee_atoms,
            (CASE
-                WHEN sell_price IS NOT NULL THEN
-                    CASE
-                        WHEN buy_price IS NOT NULL and buy_price * units_bought > sell_price * units_sold
-                            then buy_price * units_bought * fee / units_sold
```
Comment (bh2smith) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:211]:
```
buy_price * units_bought * fee / units_sold
1.001076 * 0.005 * 0.0010148996324193 / 3e-18 = 1693319440706.3
```

From this [TX](https://etherscan.io/tx/0x84d57d1d57e01dd34091c763765ddda6ff713ad67840f39735f0bf0cced11f02)

**Review by josojo — APPROVED** [review]
Looks good.

But maybe you can add a comment into the script about the "known imprecision of the formula: buy_price * units_bought * fee / units_sold" steeming from fee amounts

**Comment by bh2smith** [comment]
> maybe you can add a comment
Done!

**Comment by bh2smith** [comment]
Cc @jeff-dude

**Review by jeff-dude — APPROVED** [review]
sounds good!

not applicable to this change specifically, but noticed during review -- when you have time, do you mind revising cow models to help prep for dunesql compatibility by removing usage of `::` for casting datatypes? with the replacement being the cast() function. again, no rush! just to keep in mind moving forward.

**Comment by bh2smith** [comment]
Yes for sure I can do that and keep it in mind for all future PRs. I have started writing some thing in duneSQL but sometime they don't work the same. Also any queries depending on community sources don't work in duneSQL

**Comment by jeff-dude** [comment]
> Also any queries depending on community sources don't work in duneSQL

interesting. i'd mention that in TG to niles if that persists

**Pull request merged**